controllers/cronset_controller.go

```diff
@@ -57,15 +57,34 @@ func (r *CronSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&corev1.Node{}).
 		Watches(&source.Kind{Type: &corev1.Node{}},
 			handler.EnqueueRequestsFromMapFunc(func(node client.Object) []reconcile.Request {
+				nodeLabels := node.GetLabels()
+				r.Log.Info("Node Event", "Node", node.GetName(), "Node Labels", nodeLabels)
+
 				var cronSetObjs batchv1alpha1.CronSetList
 				_ = mgr.GetClient().List(context.TODO(), &cronSetObjs)
 
 				var requests []reconcile.Request
-				for _, obj := range cronSetObjs.Items {
+
+			CronSetLoop:
+				for _, cronSet := range cronSetObjs.Items {
```

Comment:
I thought I could use ListOption for `mgr.GetClient().List(context.TODO(), &cronSetObjs)`, but I can't get the cronset correctly because it is based on all labels in the node.
This approach is correct!

Reply:
yes. I thought same as you at first.